mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -44,6 +44,7 @@ import org.apache.spark.mllib.linalg.CholeskyDecomposition
 import org.apache.spark.mllib.optimization.NNLS
 import org.apache.spark.rdd.RDD
 import org.apache.spark.sql.{DataFrame, Dataset}
+import org.apache.spark.sql.catalyst.plans._
```

Comment:
When I first read the JIRA, I thought you wanted to provide some kind of enums without fixing APIs at all. `catalyst` is an internal API. it's not supposed to be a user face interface as well.

Reply:
yep, initially I created an enum and was using that. But, later someone pointed out in JIRA that we already have a JoinType class which we can reuse here.
Is it fine if I use a enum here?
(The motive is to have a standardised join types and detect the invalid join types at compile time itself)